src/app/GitExtUtils/GitUI/ComboBoxExtensions.cs

```diff
@@ -21,37 +18,73 @@ public static void AdjustWidthToFitContent(this ComboBox comboBox)
 
         public static void ResizeDropDownWidth(this ComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
             int calculatedWidth = GetPreferredDropDownWidth(comboBox);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
         public static void ResizeDropDownWidth(this ToolStripComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
-            int calculatedWidth = GetPreferredDropDownWidth(comboBox.Control);
+            int calculatedWidth = GetPreferredDropDownWidth((ComboBox)comboBox.Control);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
-        private static int GetPreferredDropDownWidth(dynamic comboBox)
+        private static int GetPreferredDropDownWidth(ComboBox comboBox)
         {
             int calculatedWidth = 0;
-            using dynamic graphics = comboBox.CreateGraphics();
-            foreach (object obj in comboBox.Items)
+            using Graphics graphics = comboBox.CreateGraphics();
+
+            string displayMemberName = comboBox.DisplayMember;
+            Font font = comboBox.Font;
+
+            foreach (object item in comboBox.Items)
             {
-                dynamic area = graphics.MeasureString(obj.ToString(), comboBox.Font);
+                SizeF area = graphics.MeasureString(GetDisplayValue(displayMemberName, item), font);
                 calculatedWidth = Math.Max((int)area.Width, calculatedWidth);
             }
 
-            return calculatedWidth;
+            return calculatedWidth + CalculateVerticalScrollBarWidth(comboBox);
+        }
+
+        private static string? GetDisplayValue(string displayMemberName, object item)
+        {
+            if (displayMemberName.Length == 0)
+            {
+                return item.ToString();
+            }
+
+            System.Reflection.PropertyInfo? displayMemberProperty = item
+                .GetType()
+                .GetProperty(displayMemberName);
+
+            // When an item doesn't have a property, ComboBox falls back to ToString() - but not if the property contains null
+            if (displayMemberProperty is null)
+            {
+                return item.ToString();
+            }
+
+            return displayMemberProperty
+                .GetValue(item)
+                ?.ToString();
+        }
+
+        private static int CalculateVerticalScrollBarWidth(ComboBox comboBox)
+        {
+            NativeMethods.COMBOBOXINFO cboInfo = NativeMethods.COMBOBOXINFO.Create();
```

Comment:
```suggestion
            NativeMethods.COMBOBOXINFO cboInfo = new();
            cboInfo.cbSize = sizeof(COMBOBOXINFO);
```

Reply:
`GetComboBoxInfo` should probably accept a pointer instead of a `ref` too....

```cs
    NativeMethods.COMBOBOXINFO* tConfig = new();
    tConfig->cbSize = sizeof(NativeMethods.COMBOBOXINFO);
```

There are examples in dotnet/winforms codebase in release/6.0 or release/7.0 (before they moved to CsWin32 definitions - something we should probably do too...).

Reply:
`sizeof(COMBOBOXINFO);` won't do
![image](https://github.com/user-attachments/assets/1e6c3451-66bc-45b9-8df8-279356a525ac)

I've proceeded with calculating it once and then
`cboInfo.cbSize = NativeMethods.COMBOBOXINFO.SizeOf;`

Reply:
I'm sorry but _"should probably accept a pointer"_ part is way over my head, I have not seen it done that way. Also there are several interop methods accepting `ref` params. I feel being pushed to do something new (and cool?) which was not done in GE before instead of continuing in the steps of patterns that are already present.
Usually "stylistic" changes, or changes in approach are done in a sweeping "lets change all the places at once" manner. Sorry if I am misunderstanding something.

Reply:
> `sizeof(COMBOBOXINFO);` won't do ![image](https://private-user-images.githubusercontent.com/483659/363726694-1e6c3451-66bc-45b9-8df8-279356a525ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzNDI0MzMsIm5iZiI6MTcyNTM0MjEzMywicGF0aCI6Ii80ODM2NTkvMzYzNzI2Njk0LTFlNmMzNDUxLTY2YmMtNDViOS04ZGY4LTI3OTM1NmE1MjVhYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwM1QwNTQyMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NDYxNjgxMTA1ODM3YTgwZTM1YzZhNDk5ZGMzNGNjMDc5Yzk1MjkzOWRkNjg5M2VjODkyZTE2OTNjY2M3M2NkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hojdjJ-TTsJQjBHnor_qTXJohZEIBfo2ZJdLiPPfgB8)
> 
> I've proceeded with calculating it once and then `cboInfo.cbSize = NativeMethods.COMBOBOXINFO.SizeOf;`

The screenshot contains the answer as well. The method must be marked as `unsafe`.

Reply:
I couldn't get it working with the `fixed` block, please help :)
_See `Interop adjustments based on PR review` commit_

![image](https://github.com/user-attachments/assets/329e81b7-b51f-409c-9926-8c5209aded3e)